homeassistant/components/recorder/db_schema.py

```diff
@@ -468,9 +470,13 @@ def shared_attrs_bytes_from_event(
         if state is None:
             return b"{}"
         domain = split_entity_id(state.entity_id)[0]
-        exclude_attrs = (
-            exclude_attrs_by_domain.get(domain, set()) | ALL_DOMAIN_EXCLUDE_ATTRS
-        )
+        exclude_attrs = set(ALL_DOMAIN_EXCLUDE_ATTRS)
+        if base_platform_attrs := exclude_attrs_by_domain.get(domain):
+            exclude_attrs |= base_platform_attrs
+        if (reg_ent := entity_registry.async_get(state.entity_id)) and (
```

Comment:
Instead of using the entity registry, we should use `homeassistant.helpers.entity.entity_sources`